### ISSUE TITLE
Keeping up with the APIv4 changelog

### ISSUE THREAD

**Issue by colemanw** [civicrm/civicrm-dev-docs#829]
APIv4 is still in rapid development and I haven't been keeping the changelog page up-to-date:
https://docs.civicrm.org/dev/en/latest/api/v4/changes/

I'm wondering if it would be possible to leverage the release notes. Each one has a section about api changes. We just need to somehow link that in here.

Any ideas @agh1 ?

**Comment by agh1** [comment]
To me, for the purposes of release notes, if separate commits and PRs are appearing in `civicrm-core`, that's fully a part of core.  There's no need for a separate changelog.

The contrasting, equally valid model would be that of a distinct extension or library with its own versioning.  The PR to core might say "Updated awesomeExtension to v1.2.3", and I'd expect there to be a changelog or equivalent for the extension somewhere detailing what's new in v1.2.3.

This incidentally came up in respond to @MegaphoneJon's comment on https://github.com/civicrm/civicrm-core/pull/17718#issuecomment-651890756 about extensions in core.  I summarized it [on Mattermost](https://chat.civicrm.org/civicrm/pl/gqduebhxuifd5rin66umtztaph), but the gist is that we can handle stuff in core like APIv4 and sequentialcreditnotes that may be an extension or have once been an extension because the commits are handled just like the rest of core.  We cannot handle extensions that are grabbed by distmaker in the course of generating tarballs, which is how the iATS extension is.

Basically, everything should be either like core (PRs to core for each change, full release notes coverage) or like a library (one core PR to update included version, which may have detailed revision history and documentation managed separately).

**Comment by colemanw** [comment]
Thanks @agh1.

APIv4 used to be an extension, but I think it's safe to call that an "alpha" phase of the api and not worry about changes from that era as no one was really using it at that point.

So the only changes we need to worry about logging are the ones from APIv4 in core.

Looking at how the release notes are structured, they all say at the top "Does this version... Alter the API?". That's great and helpful. But could we also make a section for APIv4 in the body of the notes?

**Comment by agh1** [comment]
Yeah we haven't ever had an API-specific section, even for v3, though I remember being on the fence about this when I started.  The tricky thing is that once an API hits maturity, there are relatively few API-specific issues that couldn't also be considered component-specific.  For example, is an issue with recording line items in the API an API problem or a CiviContribute problem?  Further complicating things is that often the fix for an API problem is in the BAO or other code that runs regardless of the interface.  Finally, from a practical perspective, we'll sometimes see issues described in the context that someone first noticed them even as they turn out to be broader-reaching.  If we're categorizing something based upon whether it only affects the API, that raises the stakes for determining the breadth of an issue that presents as an API issue.

So, I don't have a super-strong opinion about it, and it's not as if there aren't already some judgement calls for categorization (e.g. I try to put all payment things under CiviContribute, but sometimes an issue is initially reported from the perspective of paying for a membership or event registration).  However, I am concerned that having a separate API section (or more specifically, and APIv4 section) would lead to some inconsistency in where issues are listed.
